### DISCUSSION TITLE
Use font icons in tabs button navigation

### DISCUSSION THREAD

**Discussion by beppo-ivel** [appcelerator/titanium_mobile#12921] [Q&A] [answered]
I followed [this](https://devblog.axway.com/mobile-apps/cool-technique-to-use-vector-icons/) tutorial, to vector icon in titanium. 
1) Downloaded MaterialIcons-Regular.ttf and saved it under app/assets/android/fonts/
2) define a style in app.tss
3) Using the definition in my view tab1.xml

It works as expected. But for my tab-group the icon looks like a square. Same happened for label if the font size is about to small. 

1. How to style a tab, such that the font icon appears?
2. Is it possible to size a font icon (used as label text as well) in a different size as the label text?   

app.tss
```
'.ma' : {
  font : {
    fontFamily : 'MaterialIcons-Regular'
  }
}
'#playBtn': {
  backgroundColor: '#008f11',
	width: 50,
  height: 50,
  borderRadius: 50,
  bottom: 27,
  color: '#0d0208',
  font: {
    fontWeight: 'bold',
    fontSize: 31,
    fontFamily : 'MaterialIcons-Regular'
  }

}
```
views/tab1.xml
```
<Alloy>
  <Tab id="tab1" class="ma" font="fontSize:59" title="\ue425 Tab1">
    <Window id="window1" title="Window 1" >
      <Button id="playBtn" onClick="start" title="\ue037" />
    </Window>
  </Tab>
</Alloy>
```

![app](https://user-images.githubusercontent.com/72635809/123448280-09913f00-d5db-11eb-998b-45458b493e6b.png)

**Comment by m1ga** [comment, marked as answer]
It would be very helpful if you provide some code and a screenshot

I can only show how I create my tab icons:
```// generate icons
function generateIcon(icon) {
	var size = (icon.size == undefined) ? 30 : icon.size;
	if (OS_IOS) {
		size /= Ti.Platform.DisplayCaps.logicalDensityFactor;
	}
	var f = Ti.Filesystem.getFile(Ti.Filesystem.applicationDataDirectory, icon.filename);
	var lbl = Ti.UI.createLabel({
		text: icon.icon,
		font: {
			fontSize: size - 2,
			fontFamily: "remixicon"
		},
		color: "#888"
	});
	var view = Ti.UI.createView({
		width: size,
		height: size,
	});
	view.add(lbl);
	var img = view.toImage();
	f.write(img);
}

generateIcon({
	filename: 'icon_list.png',
	icon: "\uEEBA"
})
```

and the I use `$.tab1.icon = Ti.Filesystem.getFile(Ti.Filesystem.applicationDataDirectory, 'icon_list.png').read();`

**Reply by beppo-ivel** [reply]
thx for sharing, I edit my start post, to show how I tried to use vector icons.

**Reply by m1ga** [reply]
a tab has no fontFamily property. Also the icon should be over the text (more native look&feel) so check my example above and use the `icon` property.

**Reply by m1ga** [reply]
you could try a drawable for Android but honestly the example above is the quickest way to go. And since it renders the PNGs on the fly out of a font you won't see any difference between that and a vector object. 

If you want to use a font + icon there just create a custom bottom bar. ScrollableView for the views and a normal view with buttons at the bottom. Then you can create the button views as you like. If you click a button you scroll the the page you want to show. But `"\ue425 Tab1"` is not good either as it will display "Tab 1" in your icon font too and normally that is not what you want :)

**Reply by beppo-ivel** [reply]
_" And since it renders the PNGs on the fly out of a font you won't see any difference between that and a vector object."_
Sorry I was not reading your example properly. Now I think that I got your idea. But also I run into the following issues:

1. How to know which size exactly you have  to choose? If u define the size to small or to large, then the resolution of the icon does not fit the tab-element.
2. I'm not able to change the color no matter what I define under  `color: "#888"`. The same problem with the title color of tabgroup:

app.tss
```
'TabGroup[platform=android]': {
  titleColor: "#008f11",
}
```

**Reply by m1ga** [reply]
1. depends on your font, the size you would like to have and the system/phone: https://titaniumsdk.com/api/titanium/ui/tab.html#icon
```
If the icon provided is larger than [30 pixels (60 pixels for retina, 90 pixels for super-retina)] (https://developer.apple.com/ios/human-interface-guidelines/icons-and-images/image-size-and-resolution/), this image will be scaled to fit if used with activeIcon, and cropped at the bottom otherwise.
```
easiest solution: find a correct size and add `@2x.png` and `@3x.png` . e.g. in my example I use
```
	generateIcon({
		filename: 'icon_user@3x.png',
		icon: "\uF264",
		size: 90
	})
	generateIcon({
		filename: 'icon_list@2x.png',
		icon: "\uEEBA",
		size: 60
	})
```
for retina ios icons. But just try it yourself and find the correct values.


2. here is my style:
```
"Tab" : {
    tintColor: "#333",
    activeTintColor: '#fff'
}
"TabGroup" : {
    navBarHidden: true,
    tabsBackgroundColor: "#f00",
    activeTitleColor: "#fff",
    titleColor: "#333",
    swipeable: false,
    smoothScrollOnTabClick: false
}
"TabGroup[platform=android]" : {
    theme: "Theme.AppCompat.NoTitleBar",
    style: Titanium.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION
}
```

make sure to set the correct styles to either the Tab or TabGroup in order to set a color. And of course: use the latest SDK!

**Reply by beppo-ivel** [reply]
I also mean, no matter which color I set in the generateIcon function, the icon does not change.

**Reply by m1ga** [reply]
do you have a theme that will override it?

**Reply by beppo-ivel** [reply]
Don't know what exactly do u mean if u ask for theme. I set some styles in the app.tss nothing else:

```
'.ma' : {
  font : {
    fontFamily : 'MaterialIcons-Regular'
  }
}
'TabGroup[platform=android]': {
  backgroundColor: "#0d0208",
  titleColor: "#008f11",
  style: Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION
}
"Tab":{
  activeTitleColor:'#00ff41',
  titleColor: "#008f11"
}
"Menu": {
  color: "#008f11",
  backgroundColor: "#0d0208"
}
"Label": {
  width: Ti.UI.SIZE,
  height: Ti.UI.SIZE,
  color: "#008f11",
  font: {
	fontSize: 16
	}
}
```

**Reply by m1ga** [reply]
an Android theme/XML with color settings that might override your tss styles. But if you don't know what a theme is I guess you don't have any :smile: 

I'm a bit lost what your problem is. I can change the inactive color (icon + text) by setting 
```
"Tab" : {
    tintColor: "#333"
}
```

and the active color with:
```
"Tab" : {
    activeTintColor: '#f00'
}
```

for the active icon and
```
"TabGroup" : {
    activeTitleColor: "#fff"
}
```
for the active text.

Please create an example with your current code and a screenshot or even better a zip so I can test it.

**Reply by beppo-ivel** [reply]
daring implication ;)

the coloring by app.tss file is working now. Dont know what happend. But the property "color" in  your generateIcon function still has no effect.

**Reply by m1ga** [reply]
that is true. It is just there to generate a mid-grey icon since there was an issue with tinting in earlier SDKs. But as stated in the comment above: the whole coloring is done in TSS
